nucleus/glassfish-jul-extension/src/main/java/org/glassfish/main/jul/handler/LogCollectorHandler.java

```diff
@@ -106,7 +106,12 @@ public GlassFishLogRecord pop() {
     public List<GlassFishLogRecord> getAll() {
         final List<GlassFishLogRecord> list = new ArrayList<>(this.buffer.getSize());
         while (!this.buffer.isEmpty()) {
-            list.add(this.buffer.poll());
+            final GlassFishLogRecord record = this.buffer.poll();
+            if (record == null) {
+                // Unreachable code, but just in case.
```

Comment:
To help me understand. This can be reached if after checking `!this.buffer.isEmpty()` here, but before `this.buffer.poll()` some other party empties `buffer`.
`this.buffer` is a field in this class, that could be MT-access protected, but it's decided to not do that.
And if this pretty normal situation happens, suddenly `ISEx` is thrown.
:eyes:

Reply:
This class is used usually just in tests to catch logs and process them at some safe point, so null means that we do the check too early because logs are still coming.
Original code would just pass take what was already there, this code should "panic" and report at least some flaky test.